aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -636,19 +648,30 @@ Tensor grid_sampler_3d_cuda(const Tensor& input, const Tensor& grid,
   auto H = grid.size(2);
   auto W = grid.size(3);
   auto output = at::empty({N, input.size(1), D, H, W}, input.options());
-  int count = static_cast<int>(N * D * H * W);
+  int64_t count = N * D * H * W;
   if (count > 0) {
     AT_DISPATCH_FLOATING_TYPES_AND_HALF(input.scalar_type(), "grid_sampler_2d_cuda", [&] {
+      if (canUse32BitIndexMath(input) && canUse32BitIndexMath(grid)) {
+        grid_sampler_3d_kernel<scalar_t>
+          <<<GET_BLOCKS(count), CUDA_NUM_THREADS, 0, at::cuda::getCurrentCUDAStream()>>>(
+            static_cast<int>(count),
+            getTensorInfo<scalar_t, int>(input),
+            getTensorInfo<scalar_t, int>(grid),
+            getTensorInfo<scalar_t, int>(output),
+            static_cast<GridSamplerInterpolation>(interpolation_mode),
+            static_cast<GridSamplerPadding>(padding_mode),
+            align_corners);
+      } else {
       grid_sampler_3d_kernel<scalar_t>
         <<<GET_BLOCKS(count), CUDA_NUM_THREADS, 0, at::cuda::getCurrentCUDAStream()>>>(
           count,
-          getTensorInfo<scalar_t, int>(input),
-          getTensorInfo<scalar_t, int>(grid),
-          getTensorInfo<scalar_t, int>(output),
+          getTensorInfo<scalar_t, int64_t>(input),
+          getTensorInfo<scalar_t, int64_t>(grid),
+          getTensorInfo<scalar_t, int64_t>(output),
           static_cast<GridSamplerInterpolation>(interpolation_mode),
           static_cast<GridSamplerPadding>(padding_mode),
           align_corners);
-    });
+    }});
```

Comment:
nit: the closing `}` from the if-else statement is not in line with the indentation level of the `if`. Maybe we should break up the `}}` into two lines